### PR TITLE
Added Demo type to AnalysisFormat so demos can have descriptions.

### PR DESCRIPTION
**Pull request by usergenic**
- `AnalsysiFormat` now has a `Demo` interface for including description and url to demo.
 - [x] CHANGELOG.md has been updated

### CODE REVIEW

**Comment by usergenic** [comment]
Fixes https://github.com/Polymer/polymer-analyzer/issues/626 depends on https://github.com/Polymer/polymer-analyzer/pull/633

File: CHANGELOG.md [hunk 1]
```diff
@@ -14,6 +14,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 ## Unreleased
 
 * Fixed issue where jsdocs syntax in HTML comments was not parsed for polymer elements.
+* [BREAKING] `AnalysisFormat` has a new `Demo` interface for reporting `@demo` annotations from Jsdocs.
```
Comment (rictic) [inline CHANGELOG.md]: Mention that this is going from a string to an object with fields

File: src/test/static/analysis/demo/analysis.json [hunk 2]
```diff
@@ -13,8 +13,14 @@
...
             },
             "demos": [
-                "demo.html",
-                "demo.html?q=1#s"
+                {
+                    "description": "An amazing demo",
+                    "url": "demo.html"
+                },
+                {
+                    "description": "Demos might have quertystrings and/or search values.",
```
Comment (rictic) [inline src/test/static/analysis/demo/analysis.json]: quertystrings!

Comment (usergenic) [inline src/test/static/analysis/demo/analysis.json]: :LOL: fixed

File: CHANGELOG.md (hunk 1 above)
Comment (usergenic) [inline CHANGELOG.md]: Done!

**Pull request merged**